### PR TITLE
Load dump

### PR DESCRIPTION
**Pull request by a-arefina**
This PR introduces possibility to write and read data from hdf5 seismic cubes, that contain only several of 'cube', 'cube_h', 'cube_x' and fixes an error in `make_sgy`

What is DONE
* `make_sgy` writes only those traces, that were present in the original sgy. Before, if some traces were absent, they were filled with zero in hdf5, and it resulted in different number of traces in hdf5 and sgy.spec of the original file, that leaded to crushing when writing new sgy
* `make_sgy` can work with any of 'cube', 'cube_h', 'cube_x' that is present in given hdf5
* `make_hdf5` can create only specified cubes
* tests that checks `make_hdf5` and `make_sgy`

What is still NOT DONE
* `load_slide, __getitem__, load_crop` of  HDF5 SeismicGeometry still need all 3 cubes to be present
* test data, that is suitable to be put on github

Please have a look and say, if you need this work to be finished

### CODE REVIEW

File: seismiqb/src/geometry.py [hunk 1]
```diff
@@ -507,7 +507,8 @@ def show_amplitude_hist(self, scaler=None, bins=50, **kwargs):
 
     # Convert HDF5 to SEG-Y
     def make_sgy(self, path_hdf5=None, path_spec=None, postfix='',
-                 remove_hdf5=False, zip_result=True, path_segy=None):
+                 remove_hdf5=False, zip_result=True, path_segy=None,
+                 from_cubes='ixh', chunk_size=100):
         """ Convert HDF5 cube to SEG-Y format with current geometry spec.
 
         Parameters
```
Comment (AlexeyKozhevin) [inline seismiqb/src/geometry.py:514]: Extend docstring

File: seismiqb/src/geometry.py [hunk 2]
```diff
@@ -974,7 +1001,7 @@ def __getitem__(self, key):
         return crop
 
     # Convert SEG-Y to HDF5
-    def make_hdf5(self, path_hdf5=None, postfix=''):
+    def make_hdf5(self, path_hdf5=None, postfix='', store_meta=True, cubes='ixh'):
         """ Converts `.segy` cube to `.hdf5` format.
 
         Parameters
```
Comment (AlexeyKozhevin) [inline seismiqb/src/geometry.py:1007]: Also update docsting

File: seismiqb/src/geometry.py [hunk 3]
```diff
@@ -996,31 +1023,40 @@ def make_hdf5(self, path_hdf5=None, postfix=''):
...
             # Default projection (ilines, xlines, depth) and depth-projection (depth, ilines, xlines)
             pbar = tqdm(total=self.ilines_len + self.xlines_len, ncols=1000)
 
-            pbar.set_description(f'Converting {self.long_name}; ilines projection')
-            for i in range(self.ilines_len):
-                slide = self.load_slide(i, stable=False)
-                cube_hdf5[i, :, :] = slide.reshape(1, self.xlines_len, self.depth)
-                cube_hdf5_h[:, i, :] = slide.T
-                pbar.update()
+            if 'i' in cubes or 'h' in cubes:
```
Comment (AlexeyKozhevin) [inline seismiqb/src/geometry.py:1036]: Confused branching and descriptions.

**Review by AlexeyKozhevin — CHANGES_REQUESTED** [review]

File: seismiqb/src/tests/dump_test.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+import pytest
+import numpy as np
+
+from seismiqb import SeismicGeometry
+
+PATH = "/data/acquisition_footprints/7-2_ftp_input.sgy"
+
+def compare_segy_files(path_1, path_2):
+    """ Checks that two SEG-Y files contain exacly same traces
```
Comment (SergeyTsimfer) [inline seismiqb/src/tests/dump_test.py:13]: Typo

File: seismiqb/src/tests/dump_test.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import os
+
+import pytest
+import numpy as np
+
+from seismiqb import SeismicGeometry
+
+PATH = "/data/acquisition_footprints/7-2_ftp_input.sgy"
+
+def compare_segy_files(path_1, path_2):
```
Comment (SergeyTsimfer) [inline seismiqb/src/tests/dump_test.py:12]: Headers should also be compared

File: seismiqb/src/geometry.py [hunk 6]
```diff
@@ -533,29 +534,55 @@ def make_sgy(self, path_hdf5=None, path_spec=None, postfix='',
...
+            i_enc = {num: k for k, num in enumerate(ilines)}
+            x_enc = {num: k for k, num in enumerate(xlines)}
+
+            spec.ilines = ilines
+            spec.xlines = xlines
+
+            if 'i' in from_cubes and 'cube' in src:
+                cube_hdf5 = src['cube']
+
+                def get_traces(i, x):
```
Comment (SergeyTsimfer) [inline seismiqb/src/geometry.py:559]: Don't create inner functions: remember axis and use them later for `slice` creation

File: seismiqb/src/geometry.py [hunk 7]
```diff
@@ -533,29 +534,55 @@ def make_sgy(self, path_hdf5=None, path_spec=None, postfix='',
...
+                    return cube_hdf5[x_enc[x], :, i_enc[i]]
+            elif 'h' in from_cubes and 'cube_h' in src:
+                cube_hdf5 = src['cube_h']
+
+                def get_traces(i, x):
+                    return cube_hdf5[:, i_enc[i], x_enc[x]]
+            else:
+                raise ValueError("from cubes == " + from_cubes +
```
Comment (SergeyTsimfer) [inline seismiqb/src/geometry.py:572]: We use f-strings all over the library, so don't hesitate to do that too:)

File: seismiqb/src/geometry.py [hunk 8]
```diff
@@ -974,7 +1001,7 @@ def __getitem__(self, key):
         return crop
 
     # Convert SEG-Y to HDF5
-    def make_hdf5(self, path_hdf5=None, postfix=''):
+    def make_hdf5(self, path_hdf5=None, postfix='', store_meta=True, cubes='ixh'):
```
Comment (SergeyTsimfer) [inline seismiqb/src/geometry.py:1004]: `cubes` -> `projections`

File: seismiqb/src/cubeset.py [hunk 9]
```diff
@@ -119,6 +119,23 @@ def load_geometries(self, logs=True, **kwargs):
             if logs:
                 self.geometries[ix].log()
 
+    def add_geometries_targets(self, paths, dst='geom_targets'):
```
Comment (AlexeyKozhevin) [inline seismiqb/src/cubeset.py:122]: Add into `load` possibility of target loading (in the same manner as labels loading)

File: seismiqb/src/cubeset.py [hunk 10]
```diff
@@ -119,6 +119,23 @@ def load_geometries(self, logs=True, **kwargs):
...
+            Name of attribute to put targets in, by default 'geom_targets'
+        """
+        if not hasattr(self, dst):
+            setattr(self, dst, IndexedDict({ix: None for ix in self.indices}))
+
+        for ix in self.indices:
+            getattr(self, dst)[ix] = SeismicGeometry(paths[ix])
+
+
     def convert_to_hdf5(self, postfix=''):
```
Comment (AlexeyKozhevin) [inline seismiqb/src/cubeset.py:139]: Add `kwargs` to forward to `SeismicGeometry.make_hdf5`

File: seismiqb/src/geometry.py [hunk 11]
```diff
@@ -1142,7 +1164,7 @@ def __getitem__(self, key):
         return crop
 
     # Convert SEG-Y to HDF5
-    def make_hdf5(self, path_hdf5=None, postfix='', unsafe=True):
+    def make_hdf5(self, path_hdf5=None, postfix='', unsafe=True, store_meta=True, cubes='ixh', bar=True):
```
Comment (AlexeyKozhevin) [inline seismiqb/src/geometry.py:1167]: what do you think about renaming `cubes` to `projections`?

File: seismiqb/src/cubeset.py [hunk 12]
```diff
@@ -119,6 +119,23 @@ def load_geometries(self, logs=True, **kwargs):
             if logs:
                 self.geometries[ix].log()
 
+    def add_geometries_targets(self, paths, dst='geom_targets'):
+        """Create targets from given cubes
```
Comment (AlexeyKozhevin) [inline seismiqb/src/cubeset.py:123]:
```suggestion
        """ Create targets from given cubes
```

**Review by AlexeyKozhevin — CHANGES_REQUESTED** [review]

**Pull request closed without merging**